src/compiler/checker.ts

```diff
@@ -5603,18 +5603,29 @@ namespace ts {
             return compareTypes(getTypeOfSymbol(sourceProp), getTypeOfSymbol(targetProp));
         }
 
+        // A source signature matches a target signature if the two signatures have the same number of required,
+        // optional, and rest parameters.
+        function isMatchingSignature(source: Signature, target: Signature) {

```

Comment:
`hasCompatibleMinimumArity`? In any case, `isMatchingSignature` implies you're checking whether the signatures are entirely compatible (including type compatibility).